OpenRA.Mods.Cnc/Widgets/ModelWidget.cs

```diff
@@ -11,12 +11,13 @@
 
 using System;
 using OpenRA.Graphics;
-using OpenRA.Mods.Common.Graphics;
+using OpenRA.Mods.Cnc.Graphics;
+using OpenRA.Mods.Cnc.Traits;
 using OpenRA.Widgets;
 
-namespace OpenRA.Mods.Common.Widgets
+namespace OpenRA.Mods.Cnc.Widgets
```

Comment:
This will now crash the asset browser for mods that don't use the C&C library, despite not using voxels at all:

```
Exception of type `System.InvalidOperationException`: Cannot locate type: ModelWidget
   at OpenRA.ObjectCreator.CreateObject[T](String className, Dictionary`2 args) in OpenRA.Game/ObjectCreator.cs:line 115
   at OpenRA.WidgetLoader.NewWidget(String widgetType, WidgetArgs args) in OpenRA.Game/Widgets/WidgetLoader.cs:line 83
   at OpenRA.WidgetLoader.LoadWidget(WidgetArgs args, Widget parent, MiniYamlNode node) in OpenRA.Game/Widgets/WidgetLoader.cs:line 51
   at OpenRA.WidgetLoader.LoadWidget(WidgetArgs args, Widget parent, MiniYamlNode node) in OpenRA.Game/Widgets/WidgetLoader.cs:line 67
   at OpenRA.WidgetLoader.LoadWidget(WidgetArgs args, Widget parent, MiniYamlNode node) in OpenRA.Game/Widgets/WidgetLoader.cs:line 67
   at OpenRA.WidgetLoader.LoadWidget(WidgetArgs args, Widget parent, String w) in OpenRA.Game/Widgets/WidgetLoader.cs:line 43
   at OpenRA.Widgets.Ui.OpenWindow(String id, WidgetArgs args) in OpenRA.Game/Widgets/Widget.cs:line 68
   at OpenRA.Game.OpenWindow(String widget, WidgetArgs args) in OpenRA.Game/Game.cs:line 156
   at OpenRA.Mods.HV.Widgets.Logic.CustomMainMenuLogic.<>c__DisplayClass12_0.<.ctor>b__13() in /home/matthias/Entwicklung/OpenRA/OpenHV/OpenRA.Mods.HV/Widgets/Logic/CustomMainMenuLogic.cs:line 130
   at OpenRA.Mods.Common.Widgets.ButtonWidget.<.ctor>b__41_5(MouseInput _) in OpenRA.Mods.Common/Widgets/ButtonWidget.cs:line 82
   at OpenRA.Mods.Common.Widgets.ButtonWidget.HandleMouseInput(MouseInput mi) in OpenRA.Mods.Common/Widgets/ButtonWidget.cs:line 178
   at OpenRA.Widgets.Widget.HandleMouseInputOuter(MouseInput mi) in OpenRA.Game/Widgets/Widget.cs:line 425
   at OpenRA.Widgets.Ui.HandleInput(MouseInput mi) in OpenRA.Game/Widgets/Widget.cs:line 107
   at OpenRA.DefaultInputHandler.<>c__DisplayClass5_0.<OnMouseInput>b__0() in OpenRA.Game/Input/InputHandler.cs:line 50
   at OpenRA.Sync.RunUnsynced[T](Boolean checkSyncHash, World world, Func`1 fn) in OpenRA.Game/Sync.cs:line 192
   at OpenRA.Sync.RunUnsynced[T](World world, Func`1 fn) in OpenRA.Game/Sync.cs:line 179
   at OpenRA.DefaultInputHandler.OnMouseInput(MouseInput input) in OpenRA.Game/Input/InputHandler.cs:line 50
   at OpenRA.Platforms.Default.Sdl2Input.PumpInput(Sdl2PlatformWindow device, IInputHandler inputHandler, Nullable`1 lockedMousePosition) in OpenRA.Platforms.Default/Sdl2Input.cs:line 144
   at OpenRA.Platforms.Default.Sdl2PlatformWindow.PumpInput(IInputHandler inputHandler) in OpenRA.Platforms.Default/Sdl2PlatformWindow.cs:line 500
   at OpenRA.Renderer.EndFrame(IInputHandler inputHandler) in OpenRA.Game/Renderer.cs:line 328
   at OpenRA.Game.RenderTick() in OpenRA.Game/Game.cs:line 714
   at OpenRA.Game.Loop() in OpenRA.Game/Game.cs:line 837
   at OpenRA.Game.Run() in OpenRA.Game/Game.cs:line 869
   at OpenRA.Game.InitializeAndRun(String[] args) in OpenRA.Game/Game.cs:line 306
   at OpenRA.Launcher.Program.Main(String[] args) in OpenRA.Launcher/Program.cs:line 3
```

Reply:
The crash appears to tell that ModelWidget is defined in the chrome yaml files

Reply:
Yes, I forked https://github.com/OpenHV/OpenHV/blob/main/mods/hv/chrome/assetbrowser.yaml including the logic now.